scripts/feed.js

```diff
@@ -89,6 +89,13 @@ function Feed(feed_urls)
   {
     console.info("connected to ",portal.json.name,this.portals.length+"|"+this.queue.length);
 
+    for (var id = 0; id < r.home.portal.json.port.length; id++) {
```

Comment:
@neauoire Something like this? It seems to work with 0x0ade/tonne following 0x0ade - the original "hash" dat becomes the "hashbase" dat I set up in my portal.json.

Reply:
It depends, when you fork a site, the actual hash gets stored in dat.json(accessible as archive.url) right? We could overwrite the hash with that one, if someone gives you a hash base link, it could resolve the archive.url and save that instead.

Reply:
So, `r.home.portal.json.port[id] = portal.archive.url;` ? I thought we should prefer the user-set dat in `portal.json`. It refreshes your port list when someone moves their portal by

* making their hashbase url point to the new archive

or

* making the old archive have portal.json.dat pointing to the new archive.

Maybe I just misunderstood that though 😅

Reply:
Changed it to `portal.json.dat || portal.archive.url || portal.url` to prefer the "proper" archive URL over the original (and possibly broken) URL.

Reply:
No you're right.

Reply:
Do you think I should merge? I'm not sure how to test this

Reply:
Yeah, it was fine to merge this :) I didn't want you to merge this until I fixed maintenance clearing the port list.

To test the URL "fix on load," follow someone with their `dat` in the `portal.json` pointing to a hashbase URL. Wait until you connected to their portal. Then trigger a save. Your portals.json now lists the hashbase URL.

A minor thing I noticed just now is that `portal.archive.url` doesn't end with `/`. Replacing

`|| portal.archive.url ||`

with

`|| (portal.archive.url && (portal.archive.url + "/")) ||`

fixes it. I won't issue a PR for a single line fix, so it'd be nice if you could apply it :)

Reply:
(alternatively, if the URL ! endsWith("/"), append it)